### DISCUSSION TITLE
Requirement of default VPC and public subnet for AWS: Fargate

### DISCUSSION THREAD

**Discussion by yogi-sagar** [artilleryio/artillery#2389] [Artillery]
Hi There,
We are exploring the artillery/playwright option for our load & performance testing.
I came across the documentation that says, artillery with aws fargate requires default VPC + public subnet is mandatory.
I would like to know if this is still a requirement or there is any change in the latest version.

Is there a way to pass user defined vpc and subnet in the `artillery run-fargate` command. Most of the companies wont allow public subnet being configured for security reasons.

**Comment by hassy** [comment]
hi @yogi-sagar 👋 You can configure subnets/VPC to be used with the [`--subnet-ids`](https://www.artillery.io/docs/reference/cli/run-fargate#fargate-specific-flags) flag. If that flag is not set, Artillery will try to use the default VPC. The subnets do have to be configured to allow public IPs though so that Fargate tasks can pull Docker images from our public ECR repository.

I'm presuming you referring to this bit in the docs - https://www.artillery.io/docs/load-testing-at-scale/aws-fargate#could-not-find-public-subnets-in-default-vpc:~:text=AWS%20Regions%20should,Fargate%20to%20work. We'll update the docs to clarify that it only applies when subnets haven't been configured explicitly with `--subnet-ids`.

**Reply by yogi-sagar** [reply]
Thank you @hassy, yes that was the section I was referring to.
